bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/RuleProvider.java

```diff
@@ -0,0 +1,59 @@
+package org.eclipse.smarthome.automation.core;
+
+import java.util.Collection;
+
+import org.eclipse.smarthome.automation.Rule;
+import org.eclipse.smarthome.core.common.registry.AbstractProvider;
+import org.eclipse.smarthome.core.common.registry.ManagedProvider;
+import org.osgi.framework.BundleContext;
+
+/**
+ *
+ * @author Yordan Mihaylov - Initial Contribution
+ */
+public class RuleProvider extends AbstractProvider implements ManagedProvider {
```

Comment:
This should be renamed to ManagedRuleProvider and it should extend DefaultAbstractManagedProvider, then the implementation is almost automatically done, incl. persistence. (see e.g. ManagedItemThingLinkProvider as a reference).